pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -149,33 +124,39 @@ func (l *LocalPipelineRefResolver) GetPipeline(ctx context.Context, name string)
 }
 
 // resolvePipeline accepts an impl of remote.Resolver and attempts to
-// fetch a pipeline with given name. An error is returned if the
-// resolution doesn't work or the returned data isn't a valid
-// *v1beta1.Pipeline.
-func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string) (*v1beta1.Pipeline, *v1beta1.RefSource, error) {
+// fetch a pipeline with given name and verify the v1beta1 pipeline if trusted resources is enabled.
+// An error is returned if the resolution doesn't work, the verification fails
+// or the returned data isn't a valid *v1beta1.Pipeline.
+func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string, k8s kubernetes.Interface, verificationPolicies []*v1alpha1.VerificationPolicy) (*v1beta1.Pipeline, *v1beta1.RefSource, error) {
 	obj, refSource, err := resolver.Get(ctx, "pipeline", name)
 	if err != nil {
 		return nil, nil, err
 	}
-	pipelineObj, err := readRuntimeObjectAsPipeline(ctx, obj)
+	pipelineObj, err := readRuntimeObjectAsPipeline(ctx, obj, k8s, refSource, verificationPolicies)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to convert obj %s into Pipeline", obj.GetObjectKind().GroupVersionKind().String())
```

Comment:
This error message is still useful; can you move it to where the conversion is actually happening (line 167)? same for remote tasks

Reply:
Updated